docs/.vitepress/config.mts

```diff
@@ -32,6 +42,14 @@ export default defineConfig({
     ],
 
     sidebar: sidebar()
+  },
+
+  postRender(context) {
+    const html = context.teleports?.['#sefirot-modals'] || ''
+    context.teleports = {
+      ...context.teleports,
+      body: (context.teleports?.['body'] || '') + `<div id="sefirot-modals">${html}</div>`
+    }
```

Comment:
Doing SSR with teleports. I knew someone added it to vitepress, but I didn't know how to even use it. Didn't find any public repo on GitHub using it. So here is a hacky implementation. 🫠

- It extracts teleported html for `#sefirot-modals` and generate the element for it.
- VitePress renders `teleports.body` in `<body>`, so I'm appending the `#sefirot-modals` element so that it ends up at correct place.
- This needs to be added for empty html too so that teleport target is available when VitePress performs SPA routing and gets some element that needs it.

In dev this is added via theme, similar to histoire and test setup.